src/pymor/reductors/neural_network.py

```diff
@@ -294,24 +297,153 @@ def build_basis(self):
+            self.__auto_init(locals())
+
+
+        def _compute_layers_sizes(self, hidden_layers):
+            """Compute the number of neurons in the layers of the neural network
+            (make sure to increase the input dimension to account for the time)."""
+            return [len(self.fom.parameters) + 1,] + hidden_layers + [len(self.reduced_basis),]
+
+
+        def _build_rom(self):
+            """Construct the reduced order model."""
+            with self.logger.block('Building ROM ...'):
+                rom = NeuralNetworkInstationaryModel(self.fom.T, self.nt, self.neural_network,
+                                                     self.fom.parameters, name=f'{self.fom.name}_reduced')
+
+            return rom
+
+
+        def build_basis(self):
+            """Compute a reduced basis using proper orthogonal decomposition."""
+            with self.logger.block('Building reduced basis ...'):
+
+                # compute snapshots for POD and training of neural networks
+                with self.logger.block('Computing training snapshots ...'):
+                    U = self.fom.solution_space.empty()
+                    for mu in self.training_set:
+                        u = self.fom.solve(mu)
+                        if hasattr(self, 'nt'):
+                            assert self.nt == len(u) - 1
+                        else:
+                            self.nt = len(u) - 1
+                        U.append(u)
+
+                # compute reduced basis via POD
+                reduced_basis, svals = pod(U, modes=self.basis_size, rtol=self.rtol / 2.,
+                                           atol=self.atol / 2., l2_err=self.l2_err / 2.,
+                                           **(self.pod_params or {}))
+
+                self.training_data = []
+                for i, mu in enumerate(self.training_set):
+                    sample = self._compute_sample(mu, U[i*(self.nt+1):(i+1)*(self.nt+1)], reduced_basis)
+                    for elem in sample:
+                        self.training_data.append(elem)
+
+            # compute mean square loss
+            mean_square_loss = (sum(U.norm2()) - sum(svals**2)) / len(U)
+
+            return reduced_basis, mean_square_loss
+
+
+        def _compute_sample(self, mu, u, reduced_basis):
+            """Transform parameter and corresponding solution to tensors
+            (make sure to include the time instances in the inputs)."""
+            samples = []
+            parameters_with_time = []
+            dt = self.fom.T / self.nt
+
+            t = 0.
+            for i in range(self.nt + 1):
+                parameters_with_time.append(mu.with_(t=t))
+                t += dt
+
+            for mu, u_t in zip(parameters_with_time, u):
+                mu_tensor = torch.DoubleTensor(mu.to_numpy())
+                u_tensor = torch.DoubleTensor(reduced_basis.inner(u_t)[:,0])
+                samples.append((mu_tensor, u_tensor))
```

Comment:
This could also be rewritten as a comprehension, but maybe a bit too convoluted for some tastes:

```suggestion
            samples = [(torch.DoubleTensor(mu.to_numpy()), torch.DoubleTensor(reduced_basis.inner(u_t)[:,0]))
                       for mu, u_t in zip(parameters_with_time, u)]
```

Reply:
Done in 9b693fa.